### PR TITLE
Add XUnitLogChecker to log libraries dumps, do not enable for NativeAOT tests

### PR DESCRIPTION
**Pull request by carlossanlop**
Second attempt of https://github.com/dotnet/runtime/pull/93906 . That change had to be [reverted](https://github.com/dotnet/runtime/pull/94807) because NativeAOT tests were broken in outerloop.

I added an extra commit that changes the NativeAOT condition per @MichalStrehovsky 's suggestion in [this comment](https://github.com/dotnet/runtime/pull/93906#discussion_r1393720356).

### CODE REVIEW

**Review by hoyosjs — APPROVED** [review]
Lgtm as long as OuterLoop passes

**Comment by carlossanlop** [comment]
It failed again, but I see what's happening:

- The `Build` step for NativeAOT jobs compile the product by passing the `/p:TestNativeAot=true` property, which ensures XUnitLogChecker won't be built ([log](https://dev.azure.com/dnceng-public/public/_build/results?buildId=473030&view=logs&j=b0477d29-a3e8-5063-350d-e2bc4b70853a&t=ecd96a75-c556-5b24-6f26-fbed9e19b27d)).
  - Building the product for outerloop NativeAOT tests is done here: [eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml#L67](https://github.com/dotnet/runtime/blob/45978bd7191fcbb793dc12b162540022063551b6/eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml#L67)
- But the `Send to helix` step does not pass either of the two properties that `IsXUnitLogCheckerSupported` checks (`TestNativeAot` or the freshly added `RunNativeAotTestApps`). So naturally, the `IsXUnitLogCheckerSupported` property will be true and the tests will fail because they try to run the tool but can't find it ([log](https://dev.azure.com/dnceng-public/public/_build/results?buildId=473030&view=logs&j=b0477d29-a3e8-5063-350d-e2bc4b70853a&t=cc495a94-4eb4-54d1-d51d-36ad20b1ec33)).
  - Testing is done via `/p:testRunNamePrefixSuffix=NativeAOT_$(BuildConfig)` property is passed: [eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml#L71-L74](https://github.com/dotnet/runtime/blob/45978bd7191fcbb793dc12b162540022063551b6/eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml#L71-L74) 

I'm thinking that one option would be to pass the `/p:IsXUnitLogCheckerSupported=false` property to the test command for NativeAOT tests. I need to determine if I can just pass any extra global custom properties to the postBuildSteps parameters.

File: eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml [hunk 1]
```diff
@@ -72,6 +72,7 @@ extends:
                 parameters:
                   creator: dotnet-bot
                   testRunNamePrefixSuffix: NativeAOT_$(_BuildConfig)
+                  extraHelixArguments: /p:IsXUnitLogCheckerSupported=false
```
Comment (ViktorHofer) [inline eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml:75]: Why not flow `TestNativeAot` down to helix.yml instead of this property?

Comment (carlossanlop) [reply eng/pipelines/coreclr/runtime-nativeaot-outerloop.yml:75]: I don't know if it will have any unexpected side effects.

**Comment by carlossanlop** [comment]
@MichalStrehovsky @hoyosjs The latest commit seems to have worked: The NativeAOT runs seem to have unrelated test failures, but they are printing [the expected message](https://github.com/dotnet/runtime/pull/94868/files#diff-f7fb37056cce8903c2a743e9d9f9d6b9eee30041677827b9e0fa51d5f9378012R232), are _not_ running XUnitLogChecker, and the final returned exit code is the one from the test run itself:

- From the [`linux-arm64 Release NativeAOT_Libs` log file](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-94868-merge-d4beded83f5446c2b4/System.Net.HttpListener.Tests/1/console.26c331a6.log?helixlogtype=result):
```
./RunTests.sh: line 174:    23 Aborted                 (core dumped) ./System.Net.HttpListener.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing -xml testResults.xml $RSP_FILE
/root/helix/work/workitem/e
----- end Fri Nov 17 04:30:09 PM UTC 2023 ----- exit code 134 ----------------------------------------------------------
exit code 134 means SIGABRT Abort. Managed or native assert, or runtime check such as heap corruption, caused call to abort(). Core dumped.
ulimit -c value: unlimited
cat /proc/sys/kernel/core_pattern: /home/helixbot/dotnetbuild/dumps/core.%u.%p
cat /proc/sys/kernel/core_uses_pid: 0
cat: /proc/sys/kernel/coredump_filter: No such file or directory
cat /proc/sys/kernel/coredump_filter:
Looking around for any Linux dumps...
dmesg: read kernel buffer failed: Operation not permitted
The '__IsXUnitLogCheckerSupported' env var is not set.
+ export _commandExitCode=134
```
- From the [`linux_musl-x64 Release NativeAOT_Libs` log file](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-94868-merge-08d04944d50b4b38b2/System.Linq.Expressions.Tests/1/console.729179cc.log?helixlogtype=result):
```
----- end Fri Nov 17 16:32:15 UTC 2023 ----- exit code 139 ----------------------------------------------------------
./RunTests.sh: line 174:    23 Segmentation fault      (core dumped) ./System.Linq.Expressions.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing -xml testResults.xml $RSP_FILE
/root/helix/work/workitem/e
----- end Fri Nov 17 16:32:15 UTC 2023 ----- exit code 139 ----------------------------------------------------------
exit code 139 means SIGSEGV Illegal memory access. Deref invalid pointer, overrunning buffer, stack overflow etc. Core dumped.
ulimit -c value: unlimited
cat /proc/sys/kernel/core_pattern: /home/helixbot/dotnetbuild/dumps/core.%u.%p
cat /proc/sys/kernel/core_uses_pid: 1
cat: /proc/sys/kernel/coredump_filter: No such file or directory
cat /proc/sys/kernel/coredump_filter:
Looking around for any Linux dumps...
Looking for files matching core.* ...
dmesg: klogctl: Operation not permitted
The '__IsXUnitLogCheckerSupported' env var is not set.
+ export '_commandExitCode=139'
+ python /root/helix/work/correlation/reporter/run.py https://dev.azure.com/dnceng-public/ public 10820498 eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Im9PdmN6NU1fN3AtSGpJS2xGWHo5M3VfVjBabyJ9.eyJuYW1laWQiOiJjNzczZjJjMi01MTIwLTQyMDctYWZlMi1hZmFmMzVhOGJjMGEiLCJzY3AiOiJMb2NhdGlvblNlcnZpY2UuQ29ubmVjdCBQaXBlbGluZUNhY2hlLlJlYWRXcml0ZVJvb3RBY2Nlc3MgUmVhZEFuZFB1Ymxpc2hUZXN0OmNiYjE4MjYxLWM0OGYtNGFiYi04NjUxLThjZGNiNTQ3NDY0OSBSZWFkQW5kVXBkYXRlQnVpbGRCeVVyaTpjYmIxODI2MS1jNDhmLTRhYmItODY1MS04Y2RjYjU0NzQ2NDkvZG90bmV0L3J1bnRpbWUvMjY1OkJ1aWxkL0J1aWxkLzQ3MzkzMiIsImF1aSI6ImQzNWUyZjJkLTkxMzktNDQ0Yi1hMjI0LTQ2YzJmMjBiMjZjMCIsInNpZCI6IjIwMzg5NWZjLTM0NTMtNGIzZC1iOTE0LWY2ZDRiNzQzYTcyNSIsIkJ1aWxkSWQiOiJjYmIxODI2MS1jNDhmLTRhYmItODY1MS04Y2RjYjU0NzQ2NDk7NDczOTMyIiwiam9icmVmIjoiMDQxNTFlYjgtMTY4Ny00ODY0LTlhZTctOWI5ZWQ4MDA0MGZkOjIwZTZmNjczLTBiMGMtNWM4NS0zNzdjLTVkOTQxZDRjOGVlMyIsInBwaWQiOiJ2c3RmczovLy9CdWlsZC9CdWlsZC80NzM5MzIiLCJvcmNoaWQiOiIwNDE1MWViOC0xNjg3LTQ4NjQtOWFlNy05YjllZDgwMDQwZmQuYnVpbGQuYnVpbGRfbGludXhfbXVzbF94NjRfcmVsZWFzZV9uYXRpdmVhb3RfbGlicy5fX2RlZmF1bHQiLCJyZXBvSWRzIjoiIiwiaXNzIjoiYXBwLnZzdG9rZW4udmlzdWFsc3R1ZGlvLmNvbSIsImF1ZCI6ImFwcC52c3Rva2VuLnZpc3VhbHN0dWRpby5jb218dnNvOjZmY2M5MmU1LTczYTctNGY4OC04ZDEzLWQ5MDQ1YjQ1ZmIyNyIsIm5iZiI6MTcwMDIzMjYyMiwiZXhwIjoxNzAwMjUxODIyfQ.M1LiFYfZyXiO5O8or7rXoOXcdeIfYe6lYTDg9sE-eFqlBwAg1H0qBkAcQ2_XTNv_G3k-hMYGfFp6cjaJVDUWOsr7nxloZNaU7-0jW-1pPS4cfack8gUoEWXzt--f6Xcuhdgoq_4oaZ3hDQvJnISFb4cr5DpGByjktMw9u3Q96LDpl3uecjPF6wyP_ciG2Vc9hv4TsFLQtyZGSkx-ntvEyQyN_qeXVwwUr3PHrwHjPQSXTmlCXJspRNGQLoeQ7lKQ_sYo-2FB_5nXhO_6GcC3SxTsYoIYZRYOJiClmdItag6MZMpJD56rT7TjkXl1de8VvuftInbiCDVS1OHs5NfuMA
2023-11-17T16:32:16.370Z	INFO   	run.py	run(48)	main	Beginning reading of test results.
2023-11-17T16:32:16.371Z	INFO   	run.py	__init__(42)	read_results	Searching '/root/helix/work/workitem/e' for test results files
2023-11-17T16:32:16.371Z	INFO   	run.py	__init__(42)	read_results	Searching '/root/helix/work/workitem/uploads' for test results files
2023-11-17T16:32:16.371Z	WARNING	run.py	__init__(55)	read_results	No results file found in any of the following formats: xunit, junit, trx
2023-11-17T16:32:16.372Z	INFO   	run.py	packing_test_reporter(30)	report_results	Packing 0 test reports to '/root/helix/work/workitem/e/__test_report.json'
2023-11-17T16:32:16.372Z	INFO   	run.py	packing_test_reporter(33)	report_results	Packed 1816 bytes
+ python /root/helix/work/correlation/gen-debug-dump-docs.py -buildid 473932 -workitem System.Linq.Expressions.Tests -jobid 08d04944-d50b-4b38-b20f-46778845c041 -outdir /root/helix/work/workitem/uploads -templatedir /root/helix/work/correlation -dumpdir /home/helixbot/dotnetbuild/dumps -productver 9.0.0
gen-debug-dump-docs.py: read file: /root/helix/work/correlation/debug-dump-template.md
gen-debug-dump-docs.py: writing output file: /root/helix/work/workitem/uploads/how-to-debug-dump.md
gen-debug-dump-docs.py: done writing debug dump information
+ exit 139
+ export '_commandExitCode=139'
+ chmod -R 777 /home/helixbot/dotnetbuild/dumps
+ exit 139

[END EXECUTION]
Exit Code:139
```

**Review by agocke — CHANGES_REQUESTED** [review]
This seems likely to break user workflows, to me. Now test scripts have to be passed an additional flag, where they didn't before.

I don't think it's a good idea to introduce a new system to pass extra information about the product to the test. We already do this via customizing the RunScript today. https://github.com/dotnet/runtime/compare/main...agocke:runtime:xunitlogchecker?expand=1 shows an example of how we could encode the environment variable into the script. I think that's simpler and more reliable than the additions in this PR.

**Comment by MichalStrehovsky** [comment]
Is the issue only that we don't have a workable dotnet.exe?

Instead of piping through a flag, could we detect this when we try to run it and emit message to the log that we tried, but we couldn't run XUnitLogChecker?

**Comment by carlossanlop** [comment]
I added a new commit based on the feedback by @agocke. I rebased and force pushed because there were conflicts.
@agocke can you please let me know if you agree with my latest commit?

**Review by agocke — APPROVED** [review]
This looks directionally right to me, I think there may just be some parameter confusion around the xunitwrapper binaries.

And no YAML changes! Great!

**Comment by carlossanlop** [comment]
I manually triggered runtime-nativeaot-outerloop against my latest commit and it surfaced a build failure that consistently showed up in all the jobs. I don't think my change caused it. Unfortunately, the failure is preventing me from validating my changes, since a failure would show up _after_ building. @dotnet/ilc-contrib any ideas on what the root cause could be?

**Comment by carlossanlop** [comment]
Thanks @MichalStrehovsky for the quick fix!

The newly triggered run did not show the build error and was able to execute tests. The failures are unrelated to this PR and prove that my change worked and is not blocking nativeaot outerloop anymore:

- https://github.com/dotnet/runtime/issues/95413
- https://github.com/dotnet/runtime/issues/48798
- https://github.com/dotnet/runtime/issues/95414

**Pull request merged**